src/AppBundle/Validator/DonationFrequencyValidator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace AppBundle\Validator;
+
+use AppBundle\Donation\PayboxPaymentFrequency;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+class DonationFrequencyValidator extends ConstraintValidator
+{
+    private $frequencies;
+
+    public function __construct()
+    {
+        $this->frequencies = PayboxPaymentFrequency::DONATION_FREQUENCIES;
+        $this->frequencies[] = 1;
+    }
+
+    public function validate($value, Constraint $constraint)
+    {
+        if ((intval($value) === 0 && $value != PayboxPaymentFrequency::UNLIMITED_FREQUENCY) || !in_array(intval($value), $this->frequencies)) {
```

Comment:
Maybe you should encapsulate frequency value check in the `PayboxPaymentFrequency` class:

```php
$bool = PayboxPaymentFrequency::isValid($frequency);
```